packages/pwa-kit-extension-sdk/src/react/classes/ApplicationExtension.ts

```diff
@@ -75,4 +97,90 @@ export class ApplicationExtension<
         const {allRoutes} = params
         return allRoutes
     }
+
+    /**
+     * Called on the server to serialize the extension data that will be sent to the client.
+     *
+     * @returns SerializedExtension - The serialized extension data.
+     * @throws Error if the routes have not been loaded.
+     */
+    public serialize(): SerializedExtension {
+        if (this._cachedRoutes === null) {
+            throw new Error('Routes have not been loaded. Call getRoutes() before serializing')
+        }
+        console.log('--- serializing routes for extension', this.getName())
+        const serializedRoutes = this._cachedRoutes.map((route) => {
+            // Check if it is already serialized
+            if ('componentName' in route) {
+                return route
+            }
+            if (!route.component?.displayName) {
```

Comment:
@bendvc I noticed that in `extension-starter` and `extension-store-locator` the displayName is not set for components in `./pages`.

However I noticed that `getTemplateName()` is set but after doing a quick test it seems getTemplateName is not a property of `ComponentType`.

Would I need to make any changes to the `extension-starter` (and I guess `extension-chakra-store-locator` while i'm at it) so that a `displayName` is set by default and make it less likely that extension developers forget to set it?

Reply:
I think thats a good idea. Currently there really isn't anything stopping extensions from having duplicate named pages. If you have time, we'll want to create a ticket for addressing that. No need to worry about that now, but we should do that at some point.